### PR TITLE
add patch to parse VariableExpression

### PR DESCRIPTION
**Pull request by calvernaz**
This PR patches the `VariableExpression` parse method from airspeed to support variables with dashes (`$request.header.X-User`)

This change was tested in airspeed test set and broke only one test because `-` need spacing to be correctly interpreted. The failing test is the same include in this change set.

https://velocity.apache.org/engine/devel/vtl-reference.html#variables

### CODE REVIEW

File: localstack/services/apigateway/context.py [hunk 1]
```diff
@@ -154,7 +154,7 @@ def is_v1(self):
     def cookies(self):
```
Comment (whummer) [inline localstack/services/apigateway/context.py:154]: nit: was this change required to get the velocity functionality working? While we're at it, maybe we can also add a type hint, to make it more explicit:
```
    def cookies(self) -> Optional[List[str]]:
```

File: tests/unit/test_templating.py [hunk 2]
```diff
@@ -221,3 +221,8 @@ def test_keyset_functions(self):
         variables = {"input": {"body": body}}
         result = ApiGatewayVtlTemplate().render_vtl(template, variables)
         assert result == " b"
+
+    def test_dash_in_variable_name(self):
```
Comment (whummer) [inline tests/unit/test_templating.py:225]: Thanks for adding the modified upstream test here.

In addition, can we also add a small test that tests the actual functionality added (variable names with dashes).

Something like this seems to work: 
```
template = "$my-var-1"
result = ApiGatewayVtlTemplate().render_vtl(template, {"my-var-1": 123})
assert result == "123"
```

Variable assignment doesn't seem to work yet (which is probably fine, if we're aware of this limitation):
```
template = "#set($my-var-1 = 123) $my-var-1"
result = ApiGatewayVtlTemplate().render_vtl(template, {})
assert result == " 123"
```
(yields `airspeed.TemplateSyntaxError`)

**Review by whummer — COMMENTED** [review]
Thanks for tackling this @calvernaz ! As briefly discussed offline - would be great if we can add a small test for variable names with dashes. 👍

File: localstack/services/apigateway/context.py (hunk 1 above)
Comment (calvernaz) [reply localstack/services/apigateway/context.py:154]: Not required, but it's not correct as it stands. This was just an opportunity to fix it.

File: tests/unit/test_templating.py (hunk 2 above)
Comment (calvernaz) [reply tests/unit/test_templating.py:225]: Yep 👍 - variable assignment is a different use case, valid, but another.
Regarding the "actual" functionality test, I added downstream, I can added it here but I didn't forget 😄

**Review by whummer — APPROVED** [review]
LGTM 👍 🎉
